wdl/transforms/new-base/src/main/scala/wdl/transforms/base/ast2wdlom/AstToCommandSectionElement.scala

```diff
@@ -1,20 +1,44 @@
 package wdl.transforms.base.ast2wdlom
 
+import cats.instances.either._
+import cats.instances.list._
+import cats.syntax.traverse._
+import common.Checked
 import common.transforms.CheckedAtoB
+import common.validation.Checked._
+import org.apache.commons.text.StringEscapeUtils
 import wdl.model.draft3.elements.CommandPartElement.StringCommandPartElement
 import wdl.model.draft3.elements.{CommandPartElement, CommandSectionElement, CommandSectionLine}
 
 object AstToCommandSectionElement {
   def astToCommandSectionElement(implicit astNodeToCommandPartElement: CheckedAtoB[GenericAstNode, CommandPartElement]
                                 ): CheckedAtoB[GenericAst, CommandSectionElement] = CheckedAtoB.fromCheck { ast: GenericAst =>
 
-    ast.getAttributeAsVector[CommandPartElement]("parts") map { parts =>
+    ast.getAttributeAsVector[CommandPartElement]("parts") flatMap { parts =>
       val lines = makeLines(parts)
-      val trimmed = trimStartAndEnd(lines)
-      val commonPrefixLength = minimumStartLength(trimmed)
-      val commonPrefix = " " * commonPrefixLength
+      val trimmed = trimStartAndEndBlankLines(lines)
 
-      CommandSectionElement(stripStarts(trimmed, commonPrefix))
+      // Commands support either tabs or spaces. For the undefined case of mixed tabs and spaces, Cromwell returns an error.
+      // https://github.com/openwdl/wdl/blob/master/versions/1.0/SPEC.md#stripping-leading-whitespace
+      val leadingWhitespaceMap = leadingWhitespace(trimmed)
+      val distinctLeadingWhitespaceCharacters = leadingWhitespaceMap.mkString.distinct
+
+      val commonPrefix: Checked[String] = distinctLeadingWhitespaceCharacters.length match {
+        case 0 => "".validNelCheck
+        case 1 =>
+          (distinctLeadingWhitespaceCharacters.head.toString * leadingWhitespaceMap.map(_.length).min).validNelCheck
+        case _ =>
+          s"Cannot mix leading whitespace characters in command: [${distinctLeadingWhitespaceCharacters.map { c: Char =>
```

Comment:
I think perhaps move the `map` outside the interpolation to improve readability?